metadata-ingestion/src/datahub/ingestion/source/gc/datahub_gc.py

```diff
@@ -43,34 +55,72 @@ class DataHubGcSourceConfig(ConfigModel):
         description="Sleep between truncation monitoring.",
     )
 
+    dataprocess_cleanup: Optional[DataProcessCleanupConfig] = Field(
+        default=None,
+        description="Configuration for data process cleanup",
+    )
+
+    soft_deleted_entities_cleanup: Optional[SoftDeletedEntitiesCleanupConfig] = Field(
+        default=None,
+        description="Configuration for soft deleted entities cleanup",
+    )
+
 
 @dataclass
-class DataHubGcSourceReport(SourceReport):
+class DataHubGcSourceReport(DataProcessCleanupReport, SoftDeletedEntitiesReport):
```

Comment:
in general, I tend to prefer nesting / composition instead of inheritance

e.g.
```
class DataHubGcSourceReport
  data_process_cleanup: DataProcessCleanupReport
  soft_deleted_entities_cleanup: SoftDeletedEntitiesReport
```

same for the configs

however, I know it's a larger change so we can definitely defer it to later as well